man/fi_mr.3.md

```diff
@@ -310,6 +310,10 @@ The following apply to memory registration.
   requires registering regions passed to collective calls using the
   FI_COLLECTIVE flag.
 
+*FI_MR_RELAXED_ORDERING*
+: This mode bit indicates that the provider is allowed to reorder accesses
+  to the MR to improve performance.
```

Comment:
How is this visible to the app?  I checked the libibverbs and other found documentation, and none of it adds any clarity.

For the other FI_MR_xxx bits, when those are set, there are specific steps that an application must take to support the provider implementation.  What specific step(s) must an app take if this bit is set?

For example, libfabric doesn't define data ordering within a message and has separate message ordering semantics.  Why couldn't the verbs provider just enable this if available?

Reply:
I have found a couple of documents on the Internet where enabling RELAXED_ORDERING was suggested to improve performance, especially on AMD systems: 
- https://www.amd.com/content/dam/amd/en/documents/epyc-business-docs/performance-briefs/epyc-7002-series-processors-broadcom.pdf
- https://hpcadvisorycouncil.atlassian.net/wiki/spaces/HPCWORKS/pages/1280442391/AMD+2nd+Gen+EPYC+CPU+Tuning+Guide+for+InfiniBand+HPC?preview=%2F1280442391%2F1280409615%2Fimage-20200206-071308.png

We could enable it by default as you suggested, but I'm concerned about breaking applications that require a strict ordering of RDMA operations. 

As a reference, relaxed ordering in UCX is enabled by default if an AMD processor is detected. This feature can be disabled by the environment variable `UCX_IB_PCI_RELAXED_ORDERING=no`. Maybe we should take the same approach?

Reply:
Unless there's something specific that an app needs to do in regard to setting this, I would keep it internal.  I would assume we could use the current ordering semantics to know if the app requires RDMA ordering (e.g. FI_ORDER_RMA_WAW)